compiler/packages/babel-plugin-react-compiler/src/HIR/HIRBuilder.ts

```diff
@@ -212,12 +213,6 @@ export default class HIRBuilder {
     if (binding == null) {
       return null;
     }
-    // Check if the binding is from module scope, if so return null
-    const outerBinding =
-      this.parentFunction.scope.parent.getBinding(originalName);
-    if (binding === outerBinding) {
-      return null;
-    }
```

Comment:
It looks like `#resolveBabelBinding` is also used by `isContextIdentifier`, should we also copy the module scope logic there?